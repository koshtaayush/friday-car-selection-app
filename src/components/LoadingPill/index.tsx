import React from 'react';
import styled, { css, keyframes } from 'styled-components';

interface Props {
}

const LoadingPill: React.FC<Props> = (props) => {
    return (
        <React.Fragment>
            <PillContainer></PillContainer>
        </React.Fragment>
    )
}

export default LoadingPill

const keyframesShimmer = keyframes`
	from {
		background-position: -80vw 0;
	}
	to {
		background-position: 80vw 0;
	}
`

const shimmerAnimation = css`
    background: linear-gradient(to right, #eff1f3 4%, #e2e2e2 25%, #eff1f3 36%);
    background-size: 100vw 100%;
    animation: ${keyframesShimmer} 2s infinite linear;
`

const PillContainer = styled.div`
    width: 30%;
    margin: 0.25rem;
    height: 2rem;
    border-radius: 4px;
    ${shimmerAnimation}

    @media only screen and (max-width: 600px) {
        width: 100%;
    }
`
