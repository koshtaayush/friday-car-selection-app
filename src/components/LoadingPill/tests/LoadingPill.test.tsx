import React from 'react'
import { shallow } from 'enzyme'

import LoadingPill from '..'

describe('LoadingPill', () => {
    it('Should match the snapshot', () => {

        const component = shallow(
            <LoadingPill/>
        )

        expect(component).toMatchSnapshot()
    })
})
