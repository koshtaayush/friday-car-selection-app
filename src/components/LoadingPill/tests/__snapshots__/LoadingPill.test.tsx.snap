// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingPill Should match the snapshot 1`] = `
<Fragment>
  <styled.div />
</Fragment>
`;
